src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMIT.java

```diff
@@ -66,6 +69,43 @@ public void testCheckout() throws Exception {
         assertTrue(build.getWorkspace().child("add_file").isDirectory());
     }
 
+    @Test
+    public void testCheckoutAndPush() throws Exception {
+        project.setScm(createSCM("*/master"));
+        String uniqueMessage = UUID.randomUUID().toString();
+        Shell postScript = new Shell("git checkout master\n" +
```

Comment:
Could this be an issue? This would impact other tests that rely on the state of master.

Reply:
Yes possibly. You could cleanup afterwards by doing a force push back to the original commit hash, but that's not great either.
Ideally we would have a test rule to set up a new repository for each test. Up to you if you want to put the effort in to doing that though.